src/page_header.h

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+
+#ifndef LIBSQL_PAGE_HEADER_H
+#define LIBSQL_PAGE_HEADER_H
+
+typedef struct sqlite3_pcache_page sqlite3_pcache_page;
+typedef struct Pager Pager;
+typedef struct PgHdr PgHdr;
+typedef struct PCache PCache;
+
+/*
+** Every page in the cache is controlled by an instance of the following
+** structure.
+*/
+struct PgHdr {
+  sqlite3_pcache_page *pPage;    /* Pcache object page handle */
+  void *pData;                   /* Page data */
+  void *pExtra;                  /* Extra content */
+  PCache *pCache;                /* PRIVATE: Cache that owns this page */
+  PgHdr *pDirty;                 /* Transient list of dirty sorted by pgno */
+  Pager *pPager;                 /* The pager this page is part of */
+  unsigned int pgno;             /* Page number for this page */
+#ifdef SQLITE_CHECK_PAGES
```

Comment:
This is not a great thing to have in the header, since it changes the ABI -- now you need to ensure that the library using a libSQL `sqlite3.h` defines `-DSQLITE_CHECK_PAGES` when compiling iff libSQL was built with that define.

Given that it's dynamically linked, that may be difficult, and seems like something users will get wrong.

Reply:
It's a bug! Thanks, will follow up with a fix. In wal.h I dropped the conditional compilation, but missed this one.